src/main/java/net/neoforged/neoforge/common/data/RecipePrioritiesProvider.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.common.data;
+
+import com.google.common.collect.ImmutableList;
+import com.google.gson.JsonObject;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import net.minecraft.core.HolderLookup;
+import net.minecraft.data.CachedOutput;
+import net.minecraft.data.DataProvider;
+import net.minecraft.data.PackOutput;
+import net.minecraft.resources.ResourceLocation;
+
+public abstract class RecipePrioritiesProvider implements DataProvider {
+    private final PackOutput output;
+    private final CompletableFuture<HolderLookup.Provider> registriesLookup;
+    protected HolderLookup.Provider registries;
+    private final String modid;
+    private final Map<ResourceLocation, Integer> toSerialize = new HashMap<>();
```

Comment:
Use a stable map (e.g. LinkedHashMap or SortedMap) to ensure deterministic output files.